### DISCUSSION TITLE
hyprdots minimal installation

### DISCUSSION THREAD

**Discussion by Gigas002** [prasanthrangan/hyprdots#988] [Q&A] [answered]
I'm trying to use `hyprdots` the way, so I can only copy the `hyprdots/Configs` stuff and don't use all the other stuff in a repo. I don't want to always rely on a repository/clone it (by that I mean I should clone it only when I want to update configs, all the other time it won't cause any trouble if it doesn't exist) and prefer to just throw the configs instead of running scripts, that do stuff I don't want to happen.
Here I have two questions:

1. AFAIK, the `hyprdots/Source/arcs` is used only in `hyprdots/Configs/.config/hypr/scripts/wallbashspotify.sh`? So the whole directory can be removed, if I'm not relying on this script?
2. `hyprdots/Scripts/create_cache.sh` seems to be used both during installation and actually using the config (e.g. `swwwallbash.sh`). Shouldn't this script be separated somewhere else, since it's crucial during actual execution? Ofc I can do this myself in my fork and I'm not complaining, I just thought it would be logically good for execution-time scripts (`hyprdots/Configs/.config/hypr/scripts`) to not depend on installation-time ones. Also, personally I'm not sure it's a good idea to write these files in `.cache`, since cache can be cleared in any moment... Or maybe check for cache files existence (e.g. in `swwwallbash.sh` before run) and run the script (`create_cache.sh`) if needed cache files doesn't exist

**Comment by kRHYME7** [comment]
2. it is Separated

Also no apps will touch the .cache if it is not their directory (usually). Unless something is wrong.

**Reply by Gigas002** [reply]
Or unless user clears the .cache directory. Cache are temporary files that are usually safe to remove

**Reply by kRHYME7** [reply, marked as answer]
``` /tmp/ ``` directory for temporary files 

```  ~/.cache/``` directory for cached files. 

Imagine user removing cache for some apps. Also, removal of this cached files are not fatal for the whole functionality. ./create_cache.sh can generate it back or by just selecting wallpaper/theme. 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/d2f9648a-2ae8-46b3-ad64-bac64cfa2309)
 


example paru/yay we can remove their cache but in return we will redownload again some builds making reinstallation slower.

**Reply by Gigas002** [reply]
Well that's the point, that removal of it won't break anything, the caches will just be rebuilt again if needed,
I guess it's recreated in [swwwallbash.sh](https://github.com/prasanthrangan/hyprdots/blob/main/Configs/.config/hypr/scripts/swwwallbash.sh)? If so, then it's ok. I was just looking at calls for `create_cache.sh` and since I haven't found any in `swww`-related scripts I thought it's never rebuilt again, except for installation.

**Comment by kRHYME7** [comment]
1. No source/arc stores the tar.gz of gtk themes, Icons, Fonts, unless you decompress it first then add it to /Configs/

**Reply by Gigas002** [reply]
Which script does it then? I can't find that one, that uses that path (searched for `arcs`), except for `create_cache.sh`

**Reply by kRHYME7** [reply]
restore_fnt.sh

**Reply by Gigas002** [reply]
Ah, so it's not only about fonts. My  bad, didn't notice it

**Comment by Gigas002** [comment]
Hello, @kRHYME7 , it's stupid me again! Sorry to ping you, didn't want to pollute the repo with one more discussion on this topic.

I wonder, if the `Source/arcs` archives are replaceable by packages from `aur`? I think it would be better to require user to have some packages installed, e.g. in `custom_hypr.lst` and keep the track on these updates through package manager instead of having the need to clone the repo (plus, some themes can be duplicated in system, if already installed through package manager). If possible it would be awesome to get rid of archives completely, but I don't think all the themes have corresponding packages...

I haven't yet looked deep enough into this, though. The package installations are usually system-wide, so the files are in `/usr/share/icons`, for example, instead of `~/.icons`, but I don't think it would be a problem.
As for replacements, yet again - I haven't looked very deep into this, but that's what I found out ATM:

## Fonts

(probably it'd be better to require whole `nerd-fonts` meta package installation...)

Font_NotoSansCJK -> noto-fonts-cj
Font_MononokiNerd -> ttf-mononoki-nerd
Font_MaterialDesign -> ttf-material-design-icons-git
Font_MapleNerd -> ttf-maple
Font_JetBrainsMono -> ttf-jetbrains-mono-nerd
Font_CascadiaCove -> ttf-cascadia-code or ttf-cascadia-code-nerd

## Cursors

Cursor_BibataIce -> bibata-cursor-theme -- this one seems exactly the same, as current Bibata Ice, but, as expected, installed into `/usr/share/icons`

Also

`xcursor-simp1e` package group looks like it has a lot of useful cursors, e.g. `catpuccin-mocha` for existing themes in `hyprdots`, but without special cursors atm

**Reply by kRHYME7** [reply]
True, we can really replace those fonts in AUR. (Done it with JetBrains mono). 
But, IMO, i'ts better to keep both.  I do not know what's the intention of having this as archive. But I can only think of one, the older glyphs are changes overtime, requiring the maintainer to modify it overtime. Having an archive will fix this very issue. For users, installing it from AUR will never conflict as these archives are populated inside User's home directory.  The same goes to the GTK themes. And cursors. It's more for stability and being maintenance free.

AUR will be an option for the user. 

 Some users preferred a controlled version of packages too, as some really hate breaking updates.
